### DISCUSSION TITLE
Can't get it working, keep on getting HTTP Basic Error

### DISCUSSION THREAD

**Discussion by NovaViper** [languitar/pass-git-helper#80] [Q&A]
I keep getting this error:
```
remote: HTTP Basic: Access denied
fatal: Authentication failed for 'https://gitlab.com/NovaViper/dotfiles.git/'
```

I entered the correct username and password and this is my configs:
mapping:
```
[github.com*]
target=ProductivityTools/dev/github


[gitlab.com*]
target=ProductivityTools/dev/gitlab

[DEFAULT]
username_extractor=regex_search
regex_username=^user: +(.*)$
```

.gitignore
```
[pull]
	rebase = true
[fetch]
	prune = true
[push]
	default = simple
[credential]
	helper = !pass-git-helper $@
	useHttpPath = true
[gitlab]
	user = NovaViper
```
I'm using v1.2.0 on ArchLinux (compiled from the source)

**Comment by languitar** [comment]
Your second code listing shouldn't be in the `.gitignore` file but in the `.gitconfig` file in your home directory.

**Reply by languitar** [reply]
Ping @NovaViper

**Reply by NovaViper** [reply]
Oops! I accidentally put .gitignore instead of .gitconfig, but yeah, for some reason Git is acting up with all of my private repos, same error for all of them (except my public ones), also I can't get it where it doesn't ask for the username every time

**Reply by languitar** [reply]
You can add a `-l` to the pass-git-helper invocation in the `.gitconfig` file to get full debug logging. That should give a hint on what is happening. But beware: it might contain sensitive information that you have to anonymize before posting things here.

```ini
[credential]
	helper = !pass-git-helper -l $@
```

**Reply by NovaViper** [reply]
Here ya go:
```
DEBUG:root:Received request "['protocol=https\n', 'host=gitlab.com\n']"
DEBUG:root:Received action get with request:
{'protocol': 'https', 'host': 'gitlab.com'}
DEBUG:root:Parsing mapping file. Command line: None
DEBUG:root:Parsing mapping file None
DEBUG:root:Received request "{'protocol': 'https', 'host': 'gitlab.com'}"
DEBUG:root:Searching mapping to match against header "gitlab.com"
DEBUG:root:Section "gitlab.com*" matches requested header "gitlab.com"
DEBUG:root:Requesting entry "ProductivityTools/dev/gitlab" from pass
DEBUG:root:Received request "['protocol=https\n', 'host=gitlab.com\n', 'username=NovaViper\n', 'password=ProductivityTools/dev/gitlab\n']"
DEBUG:root:Received action erase with request:
{'protocol': 'https', 'host': 'gitlab.com', 'username': 'NovaViper', 'password': 'ProductivityTools/dev/gitlab'}
DEBUG:root:Parsing mapping file. Command line: None
DEBUG:root:Parsing mapping file None
INFO:root:Action erase is currently not supported
remote: HTTP Basic: Access denied
fatal: Authentication failed for 'https://gitlab.com/NovaViper/dotfiles.git/'
```

**Reply by NovaViper** [reply]
OH I figured out what was wrong, I actually named the password file the username and not 'gitlab', after renaming it, it works!

**Reply by languitar** [reply]
The erase action in the log is actually what git does when it detects that credentials are wrong. So this definitely was a problem with wrong credentials.

Btw. I would always use SSH authentication for services that offer this feature.

**Reply by NovaViper** [reply]
> 

Yeah, I eventually realized that I had to name the actual password file 'gitlab' instead of my username, I thought the target was "ProductivityTools/dev/gitlab/(password file name)", thinking gitlab was a folder not the actual file's name
